### PR TITLE
Implement Date.UTC and add a dummy implementation for Date.now

### PR DESCRIPTION
**Pull request by LaszloLango**
JerryScript-DCO-1.0-Signed-off-by: László Langó llango.u-szeged@partner.samsung.com

### CODE REVIEW

**Comment by LaszloLango** [comment]
depends on #297

File: jerry-core/ecma/builtin-objects/ecma-builtin-date.cpp [hunk 1]
```diff
@@ -38,10 +41,109 @@
...
+  }
+
+  if (args_len >= 6 && ecma_is_completion_value_empty (ret_value))
+  {
+    ECMA_TRY_CATCH (seconds_value, ecma_op_to_number (args[5]), ret_value);
+    seconds = *ecma_get_number_from_value (seconds_value);
+    ECMA_FINALIZE (seconds_value);
+  }
+
+  if (args_len == 7 && ecma_is_completion_value_empty (ret_value))
```
Comment (galpeter) [inline jerry-core/ecma/builtin-objects/ecma-builtin-date.cpp]: IMHO, if a user passes 8 arguments we should work in that case also, we just won't do anything with the 8th argument.

File: jerry-core/ecma/builtin-objects/ecma-builtin-date.cpp [hunk 2]
```diff
@@ -38,10 +41,109 @@
...
+    milliseconds = *ecma_get_number_from_value (milliseconds_value);
+    ECMA_FINALIZE (milliseconds_value);
+  }
+
+  if (ecma_is_completion_value_empty (ret_value))
+  {
+    if (!ecma_number_is_nan (year))
+    {
+      int32_t y = ecma_number_to_int32 (year);
+      if (y >=0 && y <= 99)
```
Comment (galpeter) [inline jerry-core/ecma/builtin-objects/ecma-builtin-date.cpp]: Missing space: `if (y >= 0 &&...`

File: jerry-core/ecma/builtin-objects/ecma-builtin-date.cpp (hunk 1 above)
Comment (LaszloLango) [inline jerry-core/ecma/builtin-objects/ecma-builtin-date.cpp]: Sure, I'l fix it.

File: jerry-core/ecma/builtin-objects/ecma-builtin-date.cpp [hunk 3]
```diff
@@ -63,11 +165,33 @@ ecma_builtin_date_parse (ecma_value_t this_arg, /**< this argument */
...
-  ECMA_BUILTIN_CP_UNIMPLEMENTED (this_arg, args, args_number);
+  ecma_completion_value_t ret_value = ecma_make_empty_completion_value ();
+
+  if (args_number < 2)
+  {
+    /* Note:
+     *      When the UTC function is called with fewer than two arguments,
+     *      the behaviour is implementation-dependent, so just return NaN.
+     */
+    ecma_number_t *nan = ecma_alloc_number ();
```
Comment (kkristof) [inline jerry-core/ecma/builtin-objects/ecma-builtin-date.cpp]: Probably it should be `nan_p`.

File: jerry-core/ecma/builtin-objects/ecma-builtin-date.cpp [hunk 4]
```diff
@@ -63,11 +165,33 @@ ecma_builtin_date_parse (ecma_value_t this_arg, /**< this argument */
...
+    *nan = ecma_number_make_nan ();
+    return ecma_make_normal_completion_value (ecma_make_number_value (nan));
+  }
+
+  ECMA_TRY_CATCH (time_value, ecma_date_construct_helper (args, args_number), ret_value);
+
+  ecma_number_t *time_p = ecma_get_number_from_completion_value (time_value);
+  ecma_number_t *time_clip = ecma_alloc_number ();
```
Comment (kkristof) [inline jerry-core/ecma/builtin-objects/ecma-builtin-date.cpp]: Same as above.

**Comment by kkristof** [comment]
lgtm

File: jerry-core/ecma/builtin-objects/ecma-builtin-date.cpp [hunk 5]
```diff
@@ -70,6 +70,104 @@ ecma_date_parse_date_chars (lit_utf8_iterator_t *iter, /**< iterator of the utf8
...
+
+  ecma_number_t year = *ecma_get_number_from_value (year_value);
+  ecma_number_t month = *ecma_get_number_from_value (month_value);
+  ecma_number_t date = ECMA_NUMBER_ONE;
+  ecma_number_t hours = ECMA_NUMBER_ZERO;
+  ecma_number_t minutes = ECMA_NUMBER_ZERO;
+  ecma_number_t seconds = ECMA_NUMBER_ZERO;
+  ecma_number_t milliseconds = ECMA_NUMBER_ZERO;
+
+  if (args_len >= 3 && ecma_is_completion_value_empty (ret_value))
```
Comment (egavrin) [inline jerry-core/ecma/builtin-objects/ecma-builtin-date.cpp]: Please add comment for each condition corresponding to http://www.ecma-international.org/ecma-262/5.1/#sec-15.9.3.1 bullets.

Comment (LaszloLango) [inline jerry-core/ecma/builtin-objects/ecma-builtin-date.cpp]: OK

**Comment by LaszloLango** [comment]
@egavrin, I've updated the PR, please check.

**Comment by egavrin** [comment]
Good to me

**Pull request merged**